### DISCUSSION TITLE
What is 'alias' (vs. resultKey and name)?

### DISCUSSION THREAD

**Discussion by aleksandarsusnjar** [graphql-java/graphql-java#2327] [Q&A] [answered]
I (think) I understand `name` - that is the name of the field as declared in the schema. I also (think) I understand the `resultKey` - the requested result name. i.e.:

```GraphQL
{
    something {
        resultKey: fieldName
    }
}
```

What is `alias`? Did I get `resultKey` wrong?

**Comment by aleksandarsusnjar** [comment, marked as answer]
```java
resultKey = (alias != null) ? alias : fieldName
```

**Reply by bbakerman** [reply]
yup that
